### PR TITLE
New monsters: Experimental mutant and its upgrade

### PR DESCRIPTION
**Pull request by RarkGrames**
#### Summary
```SUMMARY: Content "New monsters: Experimental mutant and its upgrade"```

#### Purpose of change
Adding two new monsters and the things needed for them to work.

#### Describe the solution
New monster, "experimental mutant" and its upgraded version, "evolved mutant". New monster flag, "REGENERATES_1"

Lorewise the mutant is one of the people that were in the first line of test subjects during the mutagen research project undergoing in the labs. These people got highly experimental, potentially dangerous mutagenic chemicals administered by the lab scientists. The experimental nature of the chemicals degraded the subjects genetic code to a level that these mutants are now halfway between a rabid animal and an institutionalized mental patient. During the various stages of mutagen research, the scientists terminated these test subjects, but when the cataclysm hit, some of the mutants managed to escape from confinement, and now they are roaming the corridors of the deserted labs. Some of them managed to escape to the wilderness.

Gameplay-wise, the experimental mutant behaves like a wild animal, harder to fight than common zombies, so it is better to not go head-to-head with it. It will kill a 0 skill, unarmored survivor. It is neutral by default, but can be made hostile very easily.

The upgrade time of the evolved mutant is 50 days. It is a hulk tier enemy, has less health and armor, but has good dodge and can leap. It is fully aggressive. This makes it a dangerous enemy if you get caught unprepared. Think of it like a feral predator hulk.

Both monsters use the "PARROT" function, like the cyborgs (constant shouting and stuff), so they can be avoided, they will flee when heavily injured and are afraid of fire. They also like meat. So it is better to avoid them or work around them then fighting them head on. They don't have the best eyesight. An evolved mutant can be dangerous to a late game survivor, even though he has a good chance to finish it in melee with survivor gear tier equipment. Of course .50 cal solves this problem too. Also, a flashbang or two can save lives, and now there is a reason to craft gas grenades and torches. Both monsters have minor regeneration, so if they get away, they might come back for round two while the player tries to stop the bleeding from his six million wounds.

These monsters add an additional challenge and difficulty to both lab starts, and day 150 lab raids.

#### Additional context
This is my first bigger scope contribution, so any review and feedback is appreciated. I tried to research everything about the monster flags and functions, but I'm not 100% sure if everything is correct.

### CODE REVIEW

**Comment by Fris0uman** [comment]
[ CR ] : Comment requested is probably better than WIP since it's finished

File: data/json/monsters/mutant.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      "SMELLS",
+      "KEENNOSE",
+      "WARM",
+      "BLEED",
+      "BASHES",
+      "PATH_AVOID_DANGER_2",
+      "REGENERATES_1",
+      "REGEN_MORALE" ]
+  }
+  },
```
Comment (Fris0uman) [inline data/json/monsters/mutant.json]: You can remove the extra `},`

**Comment by Mcrone617** [comment]
Hey @RarkGrames i'm working with @DracoGriffin to try to increase public participation in the development of this game. Specifically, I talk to devs and make posts on reddit asking people to test it out and leave feedback for them. Would you be interesting in making a downloadable link so people can test out your new monster? For changes like this it can be helpful to have 3-4 people saying it's fine balance wise, especially if you're trying to get it merged. If you're interested, all you need to do is make a downloadable build of your changes and release it on GitHub, (I can help you if you don't know how) and then i'll take it from there and make a post describing the PR, and asking people to test it. Of course, if you want too, you can post it yourself, it's whatever you prefer. Hope to hear back from you!

**Comment by RarkGrames** [comment]
@Mcrone617 That would be great actually. I don't know how to make a downloadable link out of it, so the help is appreciated.

**Comment by Mcrone617** [comment]
> @Mcrone617 That would be great actually. I don't know how to make a downloadable link out of it, so the help is appreciated.

Here's some instructions from firestorm, one of the original guys who took advantage of the PR testing.
"To author. You can upload your build directly to GitHub.

While drafting new release you can simply upload your files in archive"

https://i.imgur.com/PFVcwdc.png
So make a build, ideally up to date with the current master branch, and upload via github in the code -> release tab. let me know when it's up and i'll test it for any irregularities. hopefully that helps.

**Comment by RarkGrames** [comment]
@Mcrone617 Which files do I have to upload for making a build?

**Comment by Mcrone617** [comment]
There's a couple of options, but these two have seemingly straightforward guides on how to do it. I assume you're using windows, but there are guides for Linux and a couple of others if this doesn't work. 
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/COMPILING-VS-VCPKG.md
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/COMPILING-MSYS.md
Going here and going to your respective computer type is also helpful, so I recommend choosing one of those, and then reading this in addition to the above build. 
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/COMPILING.md#windows
Once you do that I think you can just upload the zip file into the release?

**Comment by RarkGrames** [comment]
@Mcrone617 I'm really sorry, but this is all Greek to me. I guess I have to adjust balance and bugfixes from bugreports or Reddit.

**Comment by Mcrone617** [comment]
> @Mcrone617 I'm really sorry, but this is all Greek to me. I guess I will have to adjust balance and bugfixes from bugreports or Reddit.

Fair enough, it is pretty overwhelming, and eventually you'll get feedback when it's merged into the experimental. Just more incentive for me to make a proper guide, so it's easier to do this.

**Comment by RarkGrames** [comment]
I think I've done everything I could with this. I think it is done if the checks don't fail.

**Pull request merged**